pkg/hubble/cell/config.go

```diff
@@ -134,16 +215,66 @@ func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.Int("hubble-recorder-sink-queue-size", def.RecorderSinkQueueSize, "Queue size of each Hubble recorder sink")
+	// Hubble field redaction configuration
+	flags.Bool("hubble-redact-enabled", def.EnableRedact, "Hubble redact sensitive information from flows")
+	flags.Bool("hubble-redact-http-urlquery", def.RedactHttpURLQuery, "Hubble redact http URL query from flows")
+	flags.Bool("hubble-redact-http-userinfo", def.RedactHttpUserInfo, "Hubble redact http user info from flows")
+	flags.StringSlice("hubble-redact-http-headers-allow", def.RedactHttpHeadersAllow, "HTTP headers to keep visible in flows")
+	flags.StringSlice("hubble-redact-http-headers-deny", def.RedactHttpHeadersDeny, "HTTP headers to redact from flows")
+	flags.Bool("hubble-redact-kafka-apikey", def.RedactKafkaAPIKey, "Hubble redact Kafka API key from flows")
+	// Hubble k8s v1.Events integration configuration.
+	flags.Bool("hubble-drop-events", def.EnableK8sDropEvents, "Emit packet drop Events related to pods (alpha)")
+	flags.Duration("hubble-drop-events-interval", def.K8sDropEventsInterval, "Minimum time between emitting same events")
+	flags.StringSlice("hubble-drop-events-reasons", def.K8sDropEventsReasons, "Drop reasons to emit events for")
 }
 
 func (cfg *Config) normalize() {
 	// Dynamically set the event queue size.
 	if cfg.EventQueueSize == 0 {
 		cfg.EventQueueSize = getDefaultMonitorQueueSize(runtime.NumCPU())
 	}
+	// Before moving the --hubble-drop-events-reasons flag to Config, it was
+	// registered as flags.String() and parsed through viper.GetStringSlice()
+	// in Cilium's DaemonConfig. In that case, viper is handling the split of
+	// the single string value into slice and it uses white spaces as
+	// separators. See also https://github.com/cilium/cilium/pull/33699 for
+	// more context.
+	//
+	// Since it moved to Config, the --hubble-drop-events-reasons flag is
+	// registered as flags.StringSlice() allowing multiple flag invocations,
+	// and splitting values using comma as separator (see
+	// https://pkg.go.dev/github.com/spf13/pflag#StringSlice). Since the
+	// reasons themselves have no commas nor white spaces, starting to split on
+	// commas should not introduce issues but we still need to handle white
+	// spaces splitting to maintain backward compatibility.
+	if len(cfg.K8sDropEventsReasons) == 1 {
+		cfg.K8sDropEventsReasons = strings.Fields(cfg.K8sDropEventsReasons[0])
+	}
 }
 
 func (cfg Config) validate() error {
+	if fm := cfg.ExportFieldmask; len(fm) > 0 {
+		_, err := fieldmaskpb.New(&flowpb.Flow{}, fm...)
+		if err != nil {
+			return fmt.Errorf("hubble-export-fieldmask contains invalid fieldmask '%v': %w", fm, err)
+		}
+	}
+	if len(cfg.RedactHttpHeadersAllow) > 0 && len(cfg.RedactHttpHeadersDeny) > 0 {
+		return fmt.Errorf("Only one of --hubble-redact-http-headers-allow and --hubble-redact-http-headers-deny can be specified, not both")
```

Comment:
could we use the actual `RedactHttpHeadersAllow` / `RedactHttpHeadersDeny` if the naming changes the error log might become easily outdated.